src/ui/x11/x11_ui.cpp

```diff
@@ -52,8 +52,42 @@ void UiWindowX11::_display()
             for ( size_t i = 0u; i < _point.size(); ++i ) {
                 const Point2d & point = _point[i].first;
                 XSetForeground( _uiDisplay, defaultGC, _point[i].second );
-                XDrawLine( _uiDisplay, _window, defaultGC, static_cast<int>(point.x - 1), static_cast<int>(point.y - 1),
-                           static_cast<int>(point.x + 1), static_cast<int>(point.y + 1) );
+                XDrawLine( _uiDisplay, _window, defaultGC, static_cast<int>( point.x - 1 ), static_cast<int>( point.y - 1 ), static_cast<int>( point.x + 1 ),
+                           static_cast<int>( point.y + 1 ) );
+                XDrawLine( _uiDisplay, _window, defaultGC, static_cast<int>( point.x - 1 ), static_cast<int>( point.y + 1 ), static_cast<int>( point.x + 1 ),
+                           static_cast<int>( point.y - 1 ) );
+            }
+
+            for ( size_t i = 0u; i < _lines.size(); ++i ) {
+                const Point2d & start = std::get<0>( _lines[i] );
```

Comment:
I'm thinking to combine the code with `/src/ui/win/win_ui.h` file where we use structures to store data instead of tuples. Reading such code is very difficult so we might need to use the same idea in this class.

Reply:
We could move structures from `win_ui.h` file into `src/ui/ui.h`. The only thing missing is we need an integer value instead of 3 **uint8_t** values for color.

Reply:
We could do the same as it's done for Win32: just calculate RGB value during drawing (add a function at the top of the file to do such).

Reply:
Ok, it sounds good. I'll do this.

Reply:
Great!